sharktank/sharktank/models/wan/tools/export_all.py

```diff
@@ -0,0 +1,235 @@
+import argparse
+import logging
+import time
+from typing import Dict, Any, Optional
+from pathlib import Path
+from dataclasses import fields
+import functools
+import os
+
+import torch
+import torch.nn.functional as F
+from iree.turbine.aot import *
+from iree.turbine import aot
+import numpy as np
+from sharktank.types.theta import torch_module_to_theta, Dataset
+from sharktank.transforms.dataset import set_float_dtype
+
+
+# Set a seed for reproducibility
+torch.random.manual_seed(0)
+
+# --- Helper Functions and Classes ---
+
+
+def filter_properties_for_config(
+    properties: Dict[str, Any], config_class: Any
+) -> Dict[str, Any]:
+    """Filter properties to only include fields valid for the given config class.
+
+    Args:
+        properties: Properties dictionary
+        config_class: The dataclass to filter properties for
+
+    Returns:
+        Filtered properties dictionary with only valid fields for the config class
+    """
+    # Start with hparams if available
+    if "hparams" in properties:
+        props = properties["hparams"]
+    else:
+        props = properties
+
+    # Get set of valid field names for the config class
+    valid_fields = {f.name for f in fields(config_class)}
+
+    # Filter to only include valid fields
+    filtered_props = {k: v for k, v in props.items() if k in valid_fields}
+
+    return filtered_props
+
+
+def save_dataset(path, model):
+    """Saves model parameters to a dataset file."""
+    theta = torch_module_to_theta(model)
+    theta.rename_tensors_to_paths()
+    theta.transform(functools.partial(set_float_dtype, dtype=torch.bfloat16))
```

Comment:
This is hardcoding the dtype and this is not implied by the somewhat general name of the function.